napari/layers/points/points.py

```diff
@@ -1009,7 +1073,7 @@ def selected_data(self, selected_data):
         )
 
         # Update properties based on selected points
-        if len(self._selected_data) == 0:
+        if not self._selected_data:
```

Comment:
I think the previous approach is more readable. Plus, relying on implicit bool conversion can often yield unexpected results.

Reply:
Nine lives above it is explicity created as set `self._selected_data = set(selected_data)`.
In general bool conversion in a well-written container should be no slower than `len` and in some cases should be faster. Because of this, it is suggested to use `if not self._selected_data` rather than `if len(self._selected_data) == 0:` but I'm not tied with this change.